src/Avalonia.Controls/TextBox.cs

```diff
@@ -345,7 +354,7 @@ private void HandleTextInput(string input)
                 input = RemoveInvalidCharacters(input);
                 string text = Text ?? string.Empty;
                 int caretIndex = CaretIndex;
-                if (!string.IsNullOrEmpty(input))
+                if (!string.IsNullOrEmpty(input) && (MaxLength == 0 || input.Length + text.Length - (Math.Abs(SelectionStart - SelectionEnd)) <= MaxLength))
```

Comment:
I do not think use zero as the ignore value is a good way. Maybe some developer who do not read document want to make the TextBox not support input and he will find set MaxLength is useless.

Can we change `uint` to `uint?` and check the null value as the ignore value?